difference_map/difference_map.py

```diff
@@ -118,22 +119,42 @@
         )
 
     # variable and related things
-    var_field1 = getattr(data1, var["name"])
-    var_dims1 = var_field1.dims
-    values1 = var_field1.values
-
-    # variable and related things
-    var_field2 = getattr(data2, var["name"])
-    var_dims2 = var_field2.dims
-    values2 = var_field2.values
-
-    # Check if height dimension exists
-    height_ind = [i for i, s in enumerate(var_dims1) if "height" in s]
-    if height_ind:
-        values_red1 = values1[:, var["height"], :].squeeze()
-    height_ind = [i for i, s in enumerate(var_dims2) if "height" in s]
-    if height_ind:
-        values_red2 = values2[:, var["height"], :].squeeze()
+    values1 = data1[var["name"]]
+    values2 = data2[var["name"]]
+    # Check if variable has height as dimension and if the length of the dim is >1
+    if (
+        var["zname"] in data1[var["name"]].dims
+        and data1[var["name"]].sizes[var["zname"]] > 1
+    ):
+        values_red1 = (
+            values1.isel({var["zname"]: var["height"]}).squeeze(dim=var["zname"]).values
+        )
+    else:
+        print(
+            "Warning: The variable "
+            + var["name"]
+            + " doesn't have the height dimension "
+            + var["zname"]
+            + ". Ignore this warning for 2D variables."
+        )
+        values_red1 = values1.values
+
+    if (
+        var["zname"] in data2[var["name"]].dims
+        and data2[var["name"]].sizes[var["zname"]] > 1
+    ):
+        values_red2 = (
+            values2.isel({var["zname"]: var["height"]}).squeeze(dim=var["zname"]).values
+        )
+    else:
+        print(
+            "Warning: The variable "
+            + var["name"]
+            + " doesn't have the height dimension "
+            + var["zname"]
+            + ". Ignore this warning for 2D variables."
+        )
+        values_red2 = values2.values
```

Comment:
I think it might be nicer to use a dictionary to store the various values of file 1 and file 2, and then we can just loop through the 2 files, reducing the code duplication of opening both files, selecting the height of both datasets etc?

Reply:
Thank you, I agree! I also added some checks to make sure we have the same dimensions in both files.